### PR TITLE
Added ImageObjectDetector class

### PR DESCRIPTION
**Pull request by achaar**
### Details of the Pull Request

Added ImageObjectDetector class.

Transforming the data:
```python
(train_dataset, val_dataset), dataset_info = tfds.load(
    "coco/2017", split=["train", "validation"], with_info=True, data_dir=None
)
def dict_to_array(sample):
  image = sample["image"]
  bbox = sample["objects"]["bbox"]
  class_id = tf.cast(sample["objects"]["label"], dtype=tf.int32)
  return image, (bbox, class_id)

new_train_data = train_dataset.map(dict_to_array)
new_val_data = val_dataset.map(dict_to_array)
```

Printing the contents of the dataset:

```python
import cv2
from google.colab.patches import cv2_imshow

for item in train_dataset:
  cv2_imshow(item["image"].numpy())
  print(len(item))
  break

for item in new_train_data:
  cv2_imshow(item[0].numpy())
  print("bbox: ", item[1][0][:4])
  print("labels: ", item[1][1])
  break
```

Using ImageObjectDetector Class:
```python
import autokeras as ak
from autokeras import ImageObjectDetector

img_det = ImageObjectDetector()
img_det.fit(new_train_data, validation_data=new_val_data,epochs=10)
```

### CODE REVIEW

**Comment by achaar** [comment]
What else needs to be done in this pull request?

File: autokeras/tasks/image.py [hunk 1]
```diff
@@ -416,3 +416,121 @@ def fit(
...
+                the data should be (samples, width, height)
+                or (samples, width, height, channels).
+            y: numpy.ndarray or tensorflow.Dataset. Training data y. They are the
+                tuples of bounding boxes and their corresponding class IDs w.r.t.
+                the images in x. For example, the element_spec of a TensorFlow
+                Dataset should return below results.
+                (TensorSpec(shape=(None, None, 3), # Image shape
+                            dtype=tf.uint8,
+                            name=None),
+                 (TensorSpec(shape=(None, 4), # bbox shapes
```
Comment (haifeng-jin) [inline autokeras/tasks/image.py:500]: we need a description of the 4 values meanings.

File: autokeras/tasks/image.py [hunk 2]
```diff
@@ -416,3 +416,121 @@ def fit(
...
+
+        # Arguments
+            x: numpy.ndarray or tensorflow.Dataset. Training data x. The shape of
+                the data should be (samples, width, height)
+                or (samples, width, height, channels).
+            y: numpy.ndarray or tensorflow.Dataset. Training data y. They are the
+                tuples of bounding boxes and their corresponding class IDs w.r.t.
+                the images in x. For example, the element_spec of a TensorFlow
+                Dataset should return below results.
+                (TensorSpec(shape=(None, None, 3), # Image shape
```
Comment (haifeng-jin) [inline autokeras/tasks/image.py:497]: Only the shapes should be good enough. We don't need the tensorspecs.

File: autokeras/tasks/image.py [hunk 3]
```diff
@@ -416,3 +416,121 @@ def fit(
...
+        """Search for the best model and hyperparameters for the AutoModel.
+
+        It will search for the best model based on the performances on
+        validation data.
+
+        # Arguments
+            x: numpy.ndarray or tensorflow.Dataset. Training data x. The shape of
+                the data should be (samples, width, height)
+                or (samples, width, height, channels).
+            y: numpy.ndarray or tensorflow.Dataset. Training data y. They are the
```
Comment (haifeng-jin) [inline autokeras/tasks/image.py:493]: For np, it should be a list of numpy arrays.
For tf dataset, it should be one tf dataset.

**Review by haifeng-jin — CHANGES_REQUESTED** [review]
Thanks for the PR! Please change the file permissions with chmod.
Resolve the conflicts and pass the CI tests.

**Comment by haifeng-jin** [comment]
https://github.com/keras-team/autokeras/blob/master/autokeras/keras_layers.py#L1503
Use this for lines shouldn't be covered.

**Comment by achaar** [comment]
@googlebot I signed it!

File: autokeras/tasks/image.py [hunk 4]
```diff
@@ -416,3 +416,115 @@ def fit(
...
+        # Arguments
+            x: numpy.ndarray or tensorflow.Dataset. Training data x. The shape of
+                the data should be (samples, width, height)
+                or (samples, width, height, channels). If it's a tensorflow.Dataset
+                only x is used, and each sample has an image, and corresponding
+                (bboxes, classIDs).
+            y: numpy.ndarray. Training data y. They are the
+                tuples of bounding boxes and their corresponding class IDs w.r.t.
+                the images in x. Each bounding box is defined by 4 values
+                [ymin, xmin, ymax, xmax]. Shape of the bounding boxes should be
```
Comment (haifeng-jin) [inline autokeras/tasks/image.py:498]: We may need a better way to describe these 4 values. The user may wonder how the coordinate is built. We can discuss it during the meeting.

**Review by haifeng-jin — COMMENTED** [review]
We may also need the docstrings for the predict function.

File: autokeras/tasks/image.py (hunk 4 above)
Comment (achaar) [reply autokeras/tasks/image.py]: I have updated the description.

Comment (haifeng-jin) [reply autokeras/tasks/image.py]: Would you please include the docstrings for the predict function?

Comment (achaar) [reply autokeras/tasks/image.py]: I have included that. Let me know if any other changes are required.

**Review by haifeng-jin — APPROVED** [review]
LGTM

**Pull request merged**